### PR TITLE
Add ThreadNameFilter from Dropwizard.

### PR DESCRIPTION
**Pull request by deverton**
This changes the thread name on each request so that log messages are easier to correlate with access log entries.

### CODE REVIEW

**Comment by MarkWaka** [comment]
Is it a good idea to be renaming a thread on every request it handles? Could you use a logger MDC instead? It feels like threads should maintain their identity between requests, and using MDCs would make it easier to handle formatting of the method and uri in the logs (which is I assume what this is mainly for). It would also scale better with adding other request-identifying information like opaque IDs etc.

**Comment by deverton** [comment]
Renaming threads is a (relatively) low cost operation as it's just setting the member variable and some security manager stuff. It's not just logs that benefit from thread renaming though as thread dumps become a bit more readable.

tl;dr because Dropwizard did it :)
